app/component/ItineraryPage.js

```diff
@@ -623,16 +614,13 @@ class ItineraryPage extends React.Component {
 
     setCurrentTimeToURL(config, props.match);
 
-    if (streetHashes.includes(hash)) {
-      const viaPoints = getIntermediatePlaces(props.match.location.query);
-      // Reset streetmode selection ...
+    if (altTransitHash.includes(hash) ? secondHash : hash) {
+      // in detail view
+      // If itinerary is not found in detail view, go back to summary view
       if (
-        !isEqual(
-          // if intermediate places change
-          getIntermediatePlaces(prevProps.match.location.query),
-          viaPoints,
-        ) || // detail view has no itineraries to show
-        (!state.loadingAlt && !this.mapHashToPlan(hash)?.itineraries?.length)
+        // loadingAlt is first undefined, then true and finally false
```

Comment:
Maybe it's better to define some three state enum instead of having this fuzzy logic.

Reply:
Agreed and done.